server/jobs/project_command_output_handler_test.go

```diff
@@ -99,49 +98,34 @@ func TestProjectCommandOutputHandler(t *testing.T) {
 
 	t.Run("copies buffer to new channels", func(t *testing.T) {
 		var wg sync.WaitGroup
+		var receivedMsg string
 
 		projectOutputHandler, jobStore := createProjectCommandOutputHandler(t)
+
+		// Mocking the job store acts like populating the buffer
 		When(jobStore.Get(AnyString())).ThenReturn(&jobs.Job{
 			Output: []string{Msg},
 			Status: jobs.Processing,
 		}, nil)
 
-		// send first message to populate the buffer
-		projectOutputHandler.Send(ctx, Msg)
-		time.Sleep(10 * time.Millisecond)
-
 		ch := make(chan string)
-
-		receivedMsgs := []string{}
-
-		wg.Add(1)
-		// read from channel asynchronously
 		go func() {
 			for msg := range ch {
-				receivedMsgs = append(receivedMsgs, msg)
-
-				// we're only expecting two messages here.
-				if len(receivedMsgs) >= 2 {
-					wg.Done()
-				}
+				receivedMsg = msg
+				wg.Done()
 			}
 		}()
 
-		// register channel and backfill from buffer
-		// Note: We call this synchronously because otherwise
-		// there could be a race where we are unable to register the channel
-		// before sending messages due to the way we lock our buffer memory cache
-		projectOutputHandler.Register(ctx.JobID, ch)
+		wg.Add(1)
 
-		projectOutputHandler.Send(ctx, Msg)
+		// Register the channel and wait for msg in the buffer to be read
+		projectOutputHandler.Register(ctx.JobID, ch)
 		wg.Wait()
+
 		close(ch)
 
-		expectedMsgs := []string{Msg, Msg}
-		assert.Equal(t, len(expectedMsgs), len(receivedMsgs))
-		for i := range expectedMsgs {
-			assert.Equal(t, expectedMsgs[i], receivedMsgs[i])
-		}
+		// Assert received msg is copied from the buffer
+		assert.Equal(t, receivedMsg, Msg)
```

Comment:
how is this working without send anything to the channel?

Reply:
It is because we mock the job store to return a job: https://github.com/lyft/atlantis/blob/7774339f282b00a5c94c57e10c32eba9d1625931/server/jobs/project_command_output_handler_test.go#L105-L109
Since we're only testing that buffer is copied to new channels, I figured it'd be easier to mock the buffer being populated rather than having complex setup to send a new msg and populate the buffer.